libtransact/src/database/lmdb.rs

```diff
@@ -103,14 +105,35 @@ impl LmdbDatabase {
         let txn = lmdb::ReadTransaction::new(self.ctx.env.clone()).map_err(|err| {
             DatabaseError::ReaderError(format!("Failed to create reader: {}", err))
         })?;
-        Ok(LmdbDatabaseReader { db: self, txn })
+        Ok(LmdbDatabaseReader { db: &self, txn })
     }
 
     pub fn writer(&self) -> Result<LmdbDatabaseWriter, DatabaseError> {
         let txn = lmdb::WriteTransaction::new(self.ctx.env.clone()).map_err(|err| {
             DatabaseError::WriterError(format!("Failed to create writer: {}", err))
         })?;
-        Ok(LmdbDatabaseWriter { db: self, txn })
+        Ok(LmdbDatabaseWriter { db: &self, txn })
+    }
+}
+
+pub trait Database: Sync + Send {
```

Comment:
Database should probably also have a `clone_box` function (not unlike `state::Read`, and Implement `Clone for Box<Database>`